File: C/reverse_a_number.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    long long n;
    printf("Enter a number: ");
    if (scanf("%lld", &n) != 1) return 0;

    int sign = (n < 0) ? -1 : 1;
    long long x = llabs(n);

    long long rev = 0;
    while (x > 0) {
        rev = rev * 10 + (x % 10);
        x /= 10;
    }

    printf("Reversed number: %lld\n", (long long)sign * rev);
    return 0;
}

File: Python/reverse_a_number.py
def main():
    n = int(input("Enter a number: "))
    sign = -1 if n < 0 else 1
    n = abs(n)

    rev = 0
    while n > 0:
        rev = rev * 10 + (n % 10)
        n //= 10

    print("Reversed number:", sign * rev)


if __name__ == "__main__":
    main()

